[Feat/#5] 웨디카이브 화면 구현

dev-koh opened the pull request:
## ✨ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 사용자 UI 디자인 변경 및 추가
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제


## 🛠️ 작업내용
[ 작업한 내용을 작성해주세요 ]
- 기본 웨디카이브 화면 구현
- 스크랩된 큐레이션,웨디보드가 없는 경우 ( 빈 배열, 각각 scrappedCurationIDs,scrappedWeadyboardIDs  ) 
- 스크랩된 큐레이션, 웨디보드 있는 경우 (임시 더미 데이터, 각각 scrappedCurationIDs,scrappedWeadyboardIDs)

## 📱 스크린샷
|기능|스크린샷|
|:--:|:--:|
|GIF|<img src = "https://github.com/user-attachments/assets/c3ae3f59-42be-4210-8684-755cfcceb717" width ="250">|



## 📌 리뷰 포인트
[ 스크랩된 큐레이션,웨디보드 각각에 대해서 앱을 나가도 유지되기 위한 영속성 - 모델,뷰모델  작성 아직 고민중. (스크랩 모델, 뷰모델)
 큐레이션뷰,웨디보드뷰 작성하는 개발자분들과 소통,협력 후 스크랩 기능과 관련해서 코드 작성 예정. ]

## 관련 이슈
- Resolved: #5

dev-koh committed:
[Feat#5] 웨디카이브 기본 UI 구현(스크랩 있을때)

웨디카이브 첫 화면 구현 완료.
- 기본 웨디카이브 뷰
- 상단 바
- 인디케이터 바
- ScrollView & LazyVGrid
- Sheet ("스크랩 취소하기")
TO DO
- 스크랩" 취소할 항목 "화면 구현
- 스크랩한 큐레이션,웨디보드 없을때 화면

dev-koh committed:
[Feat #5] - 스크랩 취소 화면 추가

- DeleteView 구현
- 프로퍼티 추가
- 주석 수정

dev-koh committed:
Feat/#5

- 스크랩한 게시물 없을때 하위뷰도 구현중

dev-koh committed:
[Feat#5]-웨디카이브

- 스크랩된 큐레이션/웨디카이브 없는 경우(더미데이터 이용)
-스크랩된 큐레이션/웨디보드 없는 경우 (빈배열일때)

dev-koh commented:
파일 첨부 용량 10MB 까지만 된다고해서 스크랩한 큐레이션,웨디보드 없는 경우 스크린샷은 첨부 못하였습니다. 댓글로 첨부합니다! 

https://github.com/user-attachments/assets/30e6b92d-344e-4039-8d72-4e55a2fbcdec

dev-koh committed:
[Chore] 번들 ID 수정

com.weady.Weady 로 수정.

seo1v commented on weady/weady/Sources/Feature/Weadychive/Views/WeadychiveView.swift:
```diff
@@ -2,14 +2,415 @@
+                }
+            } label: {
+                Text("스크랩 취소하기")
+                    .foregroundStyle(.red)
```
이거 systemred로 색상 등록해놨는데 확인 한 번 해보겠습니다

dev-koh replied:
네 감사합니다 !

seo1v reviewed (approved):
스크랩 있을 때 없을 때 화면 나눠서 잘 구현되어있는 거 같아요 
추후에 뷰모델로 수정하면 될 거 같습니다

keemeoow reviewed (approved):
scrappedItemList.filter 사용해서 데이터 필터링 깔끔하게 잘하신 것 같아요. 수고하셨습니다!

dev-koh committed:
[Feat#5] PR 코멘트 반영

- "스크랩취소하기" 폰트컬러 피그마에 맞게 변경  ( extension에서 못찾아서 , 피그마 색상 코드 그대로 가져와서 사용 )
-  MVVM 패턴에 맞게 수정. (추후 큐레이션이나 웨디보드 연결시에 모델이나 뷰모델 수정 필요)
- 뷰에 있는 요소들을 모델과 뷰모델로 나누어 작성.
- WeadychiveModel 작성
- WeadychiveViewModel 작성
